### PR TITLE
Add support for running Firestore integration tests against the emulator

### PR DESCRIPTION
**Pull request by wilhuff**

### CODE REVIEW

**Comment by wilhuff** [comment]
For the moment this is merely adding the plumbing for this without really tackling the development process changes required to use it regularly. I've added instructions for how to do it.

Follow-ups in separate PRs will be to add this to travis and possibly handle starting the emulator from within the test suite.

**Comment by samtstern** [comment]
While you can always just run the JAR directly, a few of the upcoming things may be convenient for you:

1) Abe has packaged the Firebase CLI into a standalone Mac binary so you can run it without needing to install Node/NPM yourself.  You can ask him how to try it. 

2) The CLI will include a command "emulators:exec <script>" that will start the emulator, run your test script, then shut the emulator down.

File: Firestore/README.md [hunk 1]
```diff
@@ -16,6 +16,31 @@
   * In Xcode select the Firestore_IntegrationTests_iOS scheme
   * ⌘-u to build and run the integration tests
 
+### Running Integration Tests - using the Firestore Emulator
+
+Note: this does not give full coverage, but is much faster than above.
```
Comment (var-const) [inline Firestore/README.md]: Just curious, what makes it much faster?

File: Firestore/Example/Tests/Util/FSTIntegrationTestCase.mm [hunk 2]
```diff
@@ -170,6 +180,13 @@ + (NSString *)projectID {
   return defaultProjectId;
 }
 
++ (bool)isRunningAgainstEmulator {
+  if (!defaultProjectId) {
```
Comment (var-const) [inline Firestore/Example/Tests/Util/FSTIntegrationTestCase.mm]: Can you add a comment on why this lazy initialization is necessary?

Comment (wilhuff) [reply Firestore/Example/Tests/Util/FSTIntegrationTestCase.mm]: Done.

File: Firestore/README.md (hunk 1 above)
Comment (wilhuff) [reply Firestore/README.md]: Several major factors:
  * Network: talking to `localhost` vs `firestore.googleapis.com`:
      * 50x lower latency: 0.033 ms RTT vs 17 ms
      * ~500x higher bandwidth: 47 Gbps vs 75 Mbps
      * No SSL to negotiate
  * Quorum: production implements a real database that does at least one paxos round
  * Durability: The emulator doesn't touch disk

This plays out in some very different numbers for common actions:
  * Write stream handshake: 0.169 ms vs 500 ms
  * Write stream single write: 0.143 ms vs 300 ms
  * Total test suite time: 15 sec vs 200 sec

**Pull request merged**